modules/opengl/src/buffer/bufferobjectarray.cpp

```diff
@@ -67,13 +72,17 @@ BufferObjectArray::~BufferObjectArray() { glDeleteVertexArrays(1, &id_); }
 GLuint BufferObjectArray::getId() const { return id_; }
 
 void BufferObjectArray::clear() {
+    if (id_ == 0) {
+        return;
+    }
+
+    glBindVertexArray(id_);
```

Comment:
humm moving this here kind of changes the preconditions... any reason for this? are this used in many places?

Reply:
It should ensure that the correct buffer object array is bound before disabling any vertex attribute arrays of other BOAs.

Reply:
yes, but previously one had to call bind before calling clear. now one should not do that?